### PR TITLE
Purescript lexer

### PR DESCRIPTION
**Pull request by yornaath**
Purescript lexer using `purescript-parsing`.

### CODE REVIEW

**Comment by bjornkihlberg** [comment]
This solution is really nice!

File: purescript/src/Lexer.purs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import Control.Alt ((<|>))
+import Data.CodePoint.Unicode (isLetter)
+import Parsing (Parser, ParserT)
+import Parsing.String (anyCodePoint, char, string)
+import Parsing.String.Basic (number, takeWhile1)
+import Token (Token(..))
+
+illegalP :: forall a. ParserT String a Token
+illegalP = (\_ -> Illegal) <$> (anyCodePoint)
```
Comment (bjornkihlberg) [inline purescript/src/Lexer.purs:13]:
```purs
illegalP :: forall a. ParserT String a Token
illegalP = Illegal <$ anyCodePoint
```

File: purescript/src/Lexer.purs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+letP :: forall a. ParserT String a Token
+letP = (\_ -> Let) <$> (string "let")
```
Comment (bjornkihlberg) [inline purescript/src/Lexer.purs:16]:
```purs
letP :: forall a. ParserT String a Token
letP = Let <$ string "let"
```

File: purescript/src/Lexer.purs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+intP :: forall a. ParserT String a Token
+intP = Int <$> number
+
+equalP :: forall a. ParserT String a Token
+equalP = (\_ -> Equal) <$> (string "=")
```
Comment (bjornkihlberg) [inline purescript/src/Lexer.purs:22]:
```purs
equalP :: forall a. ParserT String a Token
equalP = Equal <$ string "="
```

File: purescript/src/Lexer.purs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+identP :: forall a. ParserT String a Token
+identP = Ident <$> takeWhile1 isLetter
+
+semiColonP :: forall a. ParserT String a Token
+semiColonP = (\_ -> Semicolon) <$> char ';'
```
Comment (bjornkihlberg) [inline purescript/src/Lexer.purs:28]:
```purs
semiColonP :: forall a. ParserT String a Token
semiColonP = Semicolon <$ char ';'
```

**Comment by vhladko** [comment]
@yornaath hey, this one looks good, I'll merge and add you as CODEOWNER, you can do suggested improvements in future PR, or @bjornkihlberg you can create your PR with that changes and you guys can discuss it :)

**Pull request merged**